conan_provider.cmake

```diff
@@ -445,15 +462,19 @@ macro(conan_provide_dependency method package_name)
         conan_version_check(MINIMUM ${CONAN_MINIMUM_VERSION} CURRENT ${CONAN_CURRENT_VERSION})
         message(STATUS "CMake-Conan: first find_package() found. Installing dependencies with Conan")
         conan_profile_detect_default()
-        detect_host_profile(${CMAKE_BINARY_DIR}/conan_host_profile)
+        if("autodetect" IN_LIST CONAN_HOST_PROFILE)
```

Comment:
Maybe we want to also condition de ``conan profile detect`` for ``default`` only if ``default`` is included in either of the lists? There are users there explicitly avoiding the default profile.

Reply:
I considered guarding `conan_profile_detect_default()`, but the "autodetected" profile still does `include(default)`, so it's still needed regardless. There's still a couple of PRs pending to have more confidence the autodetected profile can stand on its own. On the other hand, technically speaking, as per the profile composition logic, I think the current behaviour is equivalent to having `default;autodetected` as the value for `CONAN_HOST_PROFILE`

So maybe we can do in this PR:
* refactor `include(default)` from the autodetected profile 
* set `default;autodetected` as the default value for `CONAN_HOST_PROFILE`
* only call `conan_profile_detect_default()` if `default` is found in either `CONAN_HOST_PROFILE` OR `CONAN_BUILD_PROFILE`
☝️  this should have the same behaviour  as before this PR

In a future PR:
* change the default value of `CONAN_HOST_PROFILE` to `autodetect` (or `auto`) - assuming we can detect all relevant settings from CMake for most cases


what do you think?

Reply:
Sounds a perfect plan, I like removing the ``include()`` to start with, and doing it without breaking 👍

Reply:
Done! - gone for `auto-cmake` as the auto-generated profile name